### PR TITLE
Beekeeping 3.5

### PR DESCRIPTION
**Pull request by DeityLink**
Introducing bee species!
chill
image

:cl:
* experiment: hey I'm back.
* tweak: Slightly increased the ratio of honey found in honeycomb, compared to reagents from plants.
* tweak: Slowed down the rate at which your body metabolizes honey.
* rscadd: Added Chill Bugs, a new species of bees. One or two packets of those can be found on the Vox outpost. Those bugs imported from Vox worlds need N2 to breath and suffocate in Oxygen. They produce a honey-like substance called Chill Wax, and while it doesn't heal as efficiently as regular honey does, it has the strange property of causing people who ingest it to become unable to perform any act of violence. Incidentally, Chill Bugs live up to their name and don't get angry ever, and there are only a handful of reagents that can raise their hive's toxicity, which makes raising them much easier than raising bees, if you can set up proper atmos.
* imageadd: New sprite for the beekeeping book
* tweak : bee net renamed to bug net.

### CODE REVIEW

**Comment by DeityLink** [comment]
I've split bees.dm into several files to help a bit.
With the framework now in place, adding new bee species is actually fairly easy (you still gotta make a bunch of sprites though)

I plan to expand the framework later down the line to allow for more exotic behaviors.

**Comment by DeityLink** [comment]
The effects of chill wax persist as long as there is some in your system. so using charcoal to vomit it can be used to shorten the symptoms.

**Comment by aceedex** [comment]
Honeycreepê
Joked aside nice stuff

**Comment by Kurfursten** [comment]
>Letting your hives turn acidic by planting slipping pitchers near bees

Nice though, box has been waiting on vox bees for a long time 👍

**Comment by Durgen** [comment]
\>chill wax
nifty, bar drinks when

**Comment by SonixApache** [comment]
i love you deity

File: code/_onclick/item_attack.dm [hunk 1]
```diff
@@ -62,6 +62,11 @@ obj/item/proc/get_clamped_volume()
 	if (can_operate(M, user))        //Checks if mob is lying down on table for surgery
 		if (do_surgery(M,user,I))
 			return 1
+
+	if (user.is_pacified())
+		to_chat(user, "<span class='notice'>[pick("Like...violence...what is it even good for","Nah, you don't feel like doing that.","What did \the [M] even do to you? Chill out.")]</span>")
```
Comment (DamianX) [inline code/_onclick/item_attack.dm]: missing a question mark in the first string

File: __DEFINES/global.dm [hunk 2]
```diff
@@ -320,6 +320,8 @@ var/nanocoins_lastchange = 0
 var/speciesinit = 0
 var/minimapinit = 0
 
+var/bees_species = list()
+
```
Comment (DamianX) [inline __DEFINES/global.dm:324]: Should probably be added to the globalaccess helper but it's not super important

File: code/game/machinery/bees_apiary.dm [hunk 3]
```diff
@@ -197,10 +203,21 @@
...
-				enterHive(B)
-			N.caught_bees = list()
+				if (!species || !(queen_bees_inside || worker_bees_inside))
+					species = B.species
+				if (species == B.species)
+					enterHive(B)
+					N.caught_bees.Remove(B)
+					transferred = 1
+				else
+					transferred = 2
```
Comment (DamianX) [inline code/game/machinery/bees_apiary.dm]: these are magic numbers, consider changing them to defines

**Comment by DeityLink** [comment]
@SonixApache 
>Letting your hives turn acidic by planting slipping pitchers near bees

Won't happen. Reagents that turn a hive toxic are tied to the species that inhabit it. And Chill Bugs are much more tolerant than earth bees. Sleep Toxin won't raise the toxicity.

File: code/modules/mob/living/combat.dm [hunk 4]
```diff
@@ -89,6 +89,10 @@
 	return 1
 
 /mob/living/proc/unarmed_attack_mob(mob/living/target)
+	if(is_pacified())
+		to_chat(src, "<span class='notice'>[pick("Like...violence...what is it even good for","Nah, you don't feel like doing that.","What did \the [target] even do to you? Chill out.")]</span>")
```
Comment (DamianX) [inline code/modules/mob/living/combat.dm]: same missing question mark

**Comment by DeityLink** [comment]
Do not merge yet. There's a case I haven't checked properly. (emptying a net containing different bee species on the floor). I'll do it later today.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 5]
```diff
@@ -1290,6 +1288,28 @@
...
+	id = CHILLWAX
+	description = "A bluish wax produced by insects found on Vox worlds. Sweet to the taste, albeit trippy."
+	color = "#4C78C1"
+	alpha = 250
+	nutriment_factor = 10 * REAGENTS_METABOLISM
+	density = 1.59
+	quality = 1
+	specheatcap = 1.244
+
+/datum/reagent/honey/chillwax/on_mob_life(var/mob/living/M as mob)
```
Comment (DamianX) [inline code/modules/reagents/Chemistry-Reagents.dm]: `as mob`

File: vgstation13.dme [hunk 6]
```diff
@@ -5,6 +5,124 @@
 // END_INTERNALS
 // BEGIN_FILE_DIR
 #define FILE_DIR .
+#define FILE_DIR "code"
```
Comment (DamianX) [inline vgstation13.dme]: uh?

**Comment by Zennyization** [comment]
Finally a feature that expands upon an already existing feature/ job in an interesting way. I can see how this could be used for the better of several departments on the station.

**Comment by CptWad** [comment]
mapping (box, meta, deff, packed)

**Comment by CptWad** [comment]
actually, why didnt you just add the newbee packets to the vox seed n feed?

**Comment by PromethTK11** [comment]
I didn't touch the asteroid z level so shouldn't be an issue

**Comment by Hugh-G-Rection** [comment]
Holy crap this sounds amaz-
>Vox outpost

Motherfucker.

File: __DEFINES/global.dm (hunk 2 above)
Comment (DeityLink) [reply __DEFINES/global.dm:324]: done

**Comment by DeityLink** [comment]
Fixed a bunch of runtimes.
* New sprite for the beekeeping book
image

* bee net renamed to **bug net**.

The bug net cannot hold bees of different species simultaneously. ~~Everything should be in order for merge~~. Fuck I forgot to make sure non-angry species won't attack if you empty the net, and I'm at work now. I'll fix it tonight.

@CptWad they can make new queens through beekeeping. also those aren't seeds nor feed. The main reason however is to raise their value a bit. And yeah, that shouldn't produce any conflicts unless that PR modifies the Vox outpost.

File: code/modules/mob/mob.dm [hunk 7]
```diff
@@ -2095,8 +2095,13 @@ mob/proc/on_foot()
 				alpha = initial(alpha)
 				alphas.Remove(source_define)
 
-/mob/proc/is_pacified()
+/mob/proc/is_pacified(var/message = 0,var/target,var/weapon)
 	if (reagents.has_reagent(CHILLWAX))
+		switch (message)
+			if (1)//unarmed, melee weapon, spell
```
Comment (DamianX) [inline code/modules/mob/mob.dm]: defines

**Comment by Tehsapper** [comment]
diamond bees when

**Comment by DeityLink** [comment]
when you code them.

And it's really easy now so you have no excuse. diamond bees sound lame though. how about devil bees that feed on mobs instead of plants and produce red honey of interesting properties. I've been thinking of adding those to cult 3.0 for when NarSie rises. I want the post-NarSie world to feel like you're on Remina.

**Comment by Kurfursten** [comment]
Presumably, Diamond Bees speak all different languages and come from different backgrounds, each lending unique skills to the hive. As Code Talker puts it, a true "Super Organism".

**Comment by DeityLink** [comment]
On the other hand, I'm sure I could expand the framework even further with hive-less insects such as moskitoes, roaches, and the likes

**Comment by CptWad** [comment]
\>mosquitoes 
nice
flies are on my wishlist

**Comment by DeityLink** [comment]
Alright, ran a few more tests and fixed a last runtime; everything should be in order now.

Ready for reviews/merge.

**Comment by DeityLink** [comment]
Alright, ran a few more tests and fixed a last runtime; everything should be in order now. Plants that only produce nutriments/don't produce any reagents (such as Chicken of the Stars) are the best choice for chill bugs to pollinate if you want to maximize your production of wax.

**Ready for reviews/merge.**

File: __DEFINES/global.dm (hunk 2 above)
Comment (DamianX) [reply __DEFINES/global.dm:324]: sort of but not really, you only added it to `readglobal`, not to `writeglobal` nor to `_all_globals`
the correct way is to run `gen_globals.py`

Comment (DeityLink) [reply __DEFINES/global.dm:324]: how about now

Comment (DamianX) [reply __DEFINES/global.dm:324]: 👌

**Review by DamianX — CHANGES_REQUESTED** [review]
- split `/datum/bee/proc/Die()`'s reference clearing into `Destroy()`
- For `/mob/living/simple_animal/bee` you should null the references to `destination`, each bee in `bees`, `target`, `home`, `target_plant`, each plant in `visited_plants` and `bee_species`
- Hatcheting an apiary manually nulls references to `B.home` but the apirary's `Destroy` already takes care of this so it's unnecessary. Same thing happens in `updateHealth`.

**Review by DamianX — CHANGES_REQUESTED** [review]
Additionally
>Runtime in bees_apiary.dm, line 74: Cannot read null.common_name

>Runtime in bees_apiary.dm, line 402: Cannot execute /datum/bee/queen_bee/chill (/datum/bee/queen_bee/chill).homeCall().

>Runtime in mob.dm, line 2099: Cannot execute null.has reagent().

File: code/modules/mob/living/simple_animal/bees/bees_mob.dm [hunk 8]
```diff
@@ -95,10 +95,18 @@
...
 	if(home)
 		for (var/datum/bee/B in bees)
 			home.bees_outside_hive -= B
+		home = null
+	destination = null
+	target = null
+	target_plant = null
+	for (var/datum/bee/B in bees)
+		bees.Remove(B)
+	for (var/obj/machinery/portable_atmospherics/hydroponics/H in visited_plants)
```
Comment (DamianX) [inline code/modules/mob/living/simple_animal/bees/bees_mob.dm]: just `.Cut()` the list

**Review by DamianX — APPROVED** [review]

**Pull request merged**